.map is acting on the partition index, rather than the values

```
In [27]: s = Series([-100,-50,0,50,100],index=list('ABCDE'))

In [28]: sm = Series(list('ABCAB'),list('ABCDE'))

In [29]: sm.map(s)
Out[29]: 
A   -100
B    -50
C      0
D   -100
E    -50
dtype: int64

In [30]: ddf.from_pandas(sm,npartitions=5).compute()                                         
Out[30]: 
A    A
B    B
C    C
D    A
E    B
dtype: object

In [31]: ddf.from_pandas(sm,npartitions=5).map(s)   
Out[31]: dd.Series<elemwise-b533b55a33e686222d4aafe30c2f6c9e, divisions=('A', 'B', 'C', 'D', 'E')>

In [32]: ddf.from_pandas(sm,npartitions=5).map(s).compute()
Out[32]: 
A   -100
B    -50
C      0
D    NaN
E    NaN
dtype: float64
```

so [32] should be equiv [29]

--- a/dask/dataframe/core.py
+++ b/dask/dataframe/core.py
@@ -1132,7 +1132,14 @@ class Series(_Frame):
 
     @derived_from(pd.Series)
     def map(self, arg, na_action=None):
-        return elemwise(pd.Series.map, self, arg, na_action)
+        if not (isinstance(arg, (pd.Series, dict)) or callable(arg)):
+            raise TypeError("arg must be pandas.Series, dict or callable."
+                            " Got {0}".format(type(arg)))
+        name = 'map-' + tokenize(self, arg, na_action)
+        dsk = dict(((name, i), (pd.Series.map, k, arg, na_action)) for i, k in
+                   enumerate(self._keys()))
+        dsk.update(self.dask)
+        return Series(dsk, name, self.name, self.divisions)
 
     @derived_from(pd.Series)
     def astype(self, dtype):

--- a/dask/dataframe/tests/test_dataframe.py
+++ b/dask/dataframe/tests/test_dataframe.py
@@ -679,6 +679,13 @@ def test_assign():
 
 def test_map():
     assert eq(d.a.map(lambda x: x + 1), full.a.map(lambda x: x + 1))
+    lk = dict((v, v + 1) for v in full.a.values)
+    assert eq(d.a.map(lk), full.a.map(lk))
+    assert eq(d.b.map(lk), full.b.map(lk))
+    lk = pd.Series(lk)
+    assert eq(d.a.map(lk), full.a.map(lk))
+    assert eq(d.b.map(lk), full.b.map(lk))
+    assert raises(TypeError, lambda: d.a.map(d.b))
 
 
 def test_concat():

